tests/test_report.py

```diff
@@ -356,5 +351,50 @@ def testAddWithoutRunSummary(self):
         self.assertRegex(self.actual.message, r"^WARNING.*incomplete")
 
 
+class CompileJobSummaryTestCase(unittest.TestCase):
+    """Test compiling a job summary."""
+
+    def setUp(self):
+        self.report = WmsRunReport(
+            wms_id="1.0",
+            global_wms_id="foo#1.0",
+            path="/path/to/run",
+            label="label",
+            run="run",
+            project="dev",
+            campaign="testing",
+            payload="test",
+            operator="tester",
+            run_summary="foo:1",
+            state=WmsStates.SUCCEEDED,
+            jobs=[WmsJobReport(wms_id="1.0", name="", label="foo", state=WmsStates.SUCCEEDED)],
+            total_number_jobs=1,
+            job_state_counts={state: 1 if state == WmsStates.SUCCEEDED else 0 for state in WmsStates},
+            job_summary={
+                "foo": {state: 1 if state == WmsStates.SUCCEEDED else 0 for state in WmsStates},
+            },
+        )
+
+    def tearDown(self):
+        pass
+
+    def testSummaryExists(self):
+        result = dataclasses.replace(self.report)
+        compile_job_summary(result)
+        self.assertEqual(result, self.report)
```

Comment:
How does this check that compile_job_summary is a no-op (the next test checks that it gives the right summary)?

Reply:
Right, it doesn't. Will change.